src/Nimbus/BatchSendingExtensions.cs

```diff
@@ -9,12 +8,24 @@ public static class BatchSendingExtensions
     {
         public static Task SendAll(this IBus bus, IEnumerable<IBusCommand> busCommands)
         {
-            return Task.Run(async () => { await Task.WhenAll(busCommands.Select(e => bus.Send(e))); });
+            return Task.Run(async () =>
+                                  {
+                                      foreach (var c in busCommands)
+                                      {
+                                          await bus.Send(c);
```

Comment:
Re "can it throw?": Yep. The batch sending extensions are a bit of a stop-gap at the moment.